File: PRACTICA 5/practica_5/Estadistica.java
package PRACTICA_5;

import java.util.Scanner;
import java.util.InputMismatchException;

public class Estadistica {
    public static double[] obtenerNumeros() {
        Scanner sc = new Scanner(System.in);
        double[] numeros = new double[10];
        
        while (true) {
            System.out.println("Ingrese 10 números separados por espacios (puede usar decimales con coma):");
            boolean entradaValida = true;
            
            for (int i = 0; i < 10; i++) {
                try {
                    System.out.printf("Número %d: ", i + 1);
                    numeros[i] = sc.nextDouble();
                } catch (InputMismatchException e) {
                    entradaValida = false;
                    System.out.println("Error: Por favor ingrese un número válido (use coma para decimales)");
                    sc.next(); 
                    break;
                }
            }
            
            if (entradaValida) {
                return numeros;
            } else {
                System.out.println("Error: Ingrese exactamente 10 números válidos.");
                sc.nextLine(); 
            }
        }
    }
    
    public static double calcularPromedio(double[] numeros) {
        double suma = 0;
        for (double num : numeros) {
            suma += num;
        }
        return suma / numeros.length;
    }
    
    public static double calcularDesviacion(double[] numeros, double promedio) {
        if (numeros.length < 2) {
            throw new IllegalArgumentException("Se necesitan al menos 2 datos para calcular la desviación estándar");
        }
        double sumaDiferencias = 0;
        for (double num : numeros) {
            sumaDiferencias += Math.pow(num - promedio, 2);
        }
        return Math.sqrt(sumaDiferencias / (numeros.length - 1));
    }
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        try {
            double[] numeros = obtenerNumeros();
            double promedio = calcularPromedio(numeros);
            double desviacion = calcularDesviacion(numeros, promedio);
            System.out.printf("El promedio es %.2f%n", promedio);
            System.out.printf("La desviación estándar es %.5f%n", desviacion);
            
        } catch (IllegalArgumentException e) {
            System.out.println("Error: " + e.getMessage());
        } finally {
            sc.close();
        }
    }
}

File: PROGRAMACION II/PRACTICA 5/Estadistica.py
import math

def obtener_numeros():
    while True:
        try:
            numeros = list(map(float, input("Ingrese 10 números separados por espacios: ").split()))
            if len(numeros) == 10:
                return numeros
            else:
                print("Error: Debe ingresar exactamente 10 números.")
        except ValueError:
            print("Error: Ingrese solo números válidos.")

def calcular_promedio(numeros):
     return sum(numeros) / len(numeros)

def calcular_desviacion(numeros, promedio):
    suma_diferencias = sum((x - promedio) ** 2 for x in numeros)
    return math.sqrt(suma_diferencias / (len(numeros) - 1))

def main():
    numeros = obtener_numeros()
    promedio = calcular_promedio(numeros)
    desviacion = calcular_desviacion(numeros, promedio)

    print(f"El promedio es {promedio:.2f}")
    print(f"La desviación estándar es {desviacion:.5f}")

# Funcion principal
if __name__ == "__main__":
    main()

